tests/test_frameworks.py

```diff
@@ -0,0 +1,180 @@
+    identified_files = err.metadata.get("identified_files")
+    framework_files = err.metadata.get("framework_files")
+    assert identified_files and framework_files
+
+    for f in files:
+        assert f in identified_files
+        assert f in framework_files
+
+
+def test_skeleton_framework_identification():
+    """
+    Tests that skeleton add-ons are detected as belonging to the correct
+    framework, and rejected if applicable.
+    """
+
+    FRAMEWORKS = {
+        "besttoolbars": {
+            "unversioned": True,
+            "framework_files": {
+                "chrome/content/framework.js": "//Foo besttoolbars bar",
+                "chrome/content/config.json": {}}},
+        "conduit": {
+            "banned": True,
+            "framework_files": {
+                "searchplugins/conduit.xml": KNOWN_SCRIPT,
+                "components/ConduitFoo.js": True}},
+        "crossrider": {
+            "banned": True,
+            "framework_files": {
+                "chrome/content/CrossriderEXT.js": True,
+                "chrome/content/crossrider.js": True,
+                "chrome/content/crossriderapi.js": True}},
+        # Don't bother with Jetpack here. It has a whole file
+        # dedicated to it.
+        "kango": {
+            "framework_files": {"bootstrap.js": True,
+                                "extension_info.json": {"kango_version": "42"},
+                                "kango/foo.js": True,
+                                "kango-ui/bar.js": True}},
+        "openforge": {
+            "banned": True,
+            "framework_files": {
+                "resources/f/data/forge/all.js": True},
+            # Need to include some Jetpack-ish files to make sure
+            # this is identified as Forge even though it's built on
+            # Jetpack.
+            "other_files": {
+                "bootstrap.js": True,
+                "harness-options.json":
+                    {"sdkVersion": "1.8"}}},
+        "scriptify": {
+            "framework_files": {"bootstrap.js": True,
+                                "scriptify.json":
+                                    {"scriptify-version": "42"}}},
+    }
+
+    for framework_id, data in FRAMEWORKS.iteritems():
+        xpi = MockXPI(dict(chain(data["framework_files"].iteritems(),
+                                 data.get("other_files", {}).iteritems())))
+        err = _do_test(xpi)
+
+        framework = err.metadata["framework"]
+
+        eq_(framework["id"], framework_id)
+
+        if data.get("banned"):
+            assert (("testcases_frameworks", "banned_framework") in
+                    (e["id"] for e in err.errors))
+        else:
+            # We don't bother reporting framework files for banned
+            # frameworks, since they're blanket rejected and won't
+            # see further review.
+
+            eq_(set(data["framework_files"]),
+                set(err.metadata["framework_files"]))
+
+            if data.get("unversioned"):
```

Comment:
Hm. The version is generated from the contents of the files. I don't really want to hard code it, since that would rely on the behavior of `fake_version`, or the contents of the stub files, not changing, which I don't want to depend on. I just want to make sure that it comes out to the same as the return value for `fake_version`.